dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java

```diff
@@ -993,59 +998,68 @@ private boolean isOpenNode(String[] openNodes, String node) {
 		return returnValue;
 	}
 
+
 	// http://jira.dotmarketing.net/browse/DOTCMS-3232
 	protected Folder findSystemFolder() throws DotDataException {
-		Folder folder = new Folder();
-		folder = folderCache.getFolder(SYSTEM_FOLDER);
-		if (folder!=null && folder.getInode().equalsIgnoreCase(SYSTEM_FOLDER)) {
+		Folder folder = find(SYSTEM_FOLDER);
+		if(folder!=null && UtilMethods.isSet(folder.getInode())){
 			return folder;
-		} else {
-			folder = find(SYSTEM_FOLDER);
 		}
-		if (UtilMethods.isSet(folder.getInode()) && folder.getInode().equalsIgnoreCase(SYSTEM_FOLDER)) {
-			folderCache.addFolder(folder,APILocator.getIdentifierAPI().find(folder.getIdentifier()));
+		folder = find(OLD_SYSTEM_FOLDER_ID);
+
+		if (folder!=null && UtilMethods.isSet(folder.getInode()) ) {
 			return folder;
-		} else {
+		}
+
+
+		try(Connection connection = DbConnectionFactory.getConnection()) {
 			DotConnect dc = new DotConnect();
-			Folder folder1 = new Folder();
-			String hostInode = "";
-			folder1.setInode(SYSTEM_FOLDER);
-			folder1.setName(SYSTEM_FOLDER_ASSET_NAME);
-			folder1.setTitle("System folder");
-			try {
-				hostInode = APILocator.getHostAPI().findSystemHost(APILocator.getUserAPI().getSystemUser(), true).getIdentifier();
-			} catch (DotSecurityException e) {
-				Logger.error(FolderFactoryImpl.class, e.getMessage(), e);
-				throw new DotDataException(e.getMessage(), e);
+
+			// if we are missing the system folder identifier, we create it
+			if (dc.setSQL(
+					"select count(id) as test from identifier where asset_type='folder' and host_inode='SYSTEM_HOST' "
+							+ "and ( id = 'SYSTEM_FOLDER' or lower(parent_path) = '/system folder') ").loadInt("test",connection)
+					== 0) {
+
+				dc.setSQL("INSERT INTO IDENTIFIER(ID,PARENT_PATH,ASSET_NAME,HOST_INODE,ASSET_TYPE) VALUES(?,?,?,?,?)");
+				dc.addParam(SYSTEM_FOLDER);
+				dc.addParam(SYSTEM_FOLDER_PARENT_PATH);
+				dc.addParam("system folder");
+				dc.addParam(Host.SYSTEM_HOST);
```

Comment:
we have a constant SYSTEM_FOLDER_ASSET_NAME

Reply:
agreed.